### ISSUE TITLE
getAvailablePurchases on iOS hangs if user cancelled

### ISSUE THREAD

**Issue by mobilinked** [dooboolab/flutter_inapp_purchase#66]
### Version of flutter_inapp_purchase
^0.9.0

### Platforms you faced the error (IOS or Android or both?)
iOS 12.1

### Expected behavior
"await FlutterInappPurchase.getAvailablePurchases()" should return (null or empty) if the user tap the cancel button in the login dialog

### Actual behavior
"await FlutterInappPurchase.getAvailablePurchases()" hangs

### Tested environment (Emulator? Real Device?)
iOS simulator and real device (iPhone XS Max)

### Steps to reproduce the behavior
1. On simulator
- execute await FlutterInappPurchase.getAvailablePurchases()
- when iOS prompt to request login, tap the cancel

2. On real device
- in iOS preference, let the sandbox user be empty
- execute await FlutterInappPurchase.getAvailablePurchases()
- when iOS prompt to request login, tap the cancel

**Comment by hyochan** [comment]
We are having a few issues in `restoring` purchases in `ios`. We need the help of the community to find out the problem and willing to providing a solution.

**Comment by evanave** [comment]
I'm having issues as well.

**Comment by andymstone** [comment]
I think the problem here (and on issue #60 ) is that the iOS plugin doesn't implement the `restoreCompletedTransactionsFailedWithError` callback.  You could fix this by adding the following method in the iOS .m file:

```
-(void)paymentQueue:(SKPaymentQueue *)queue restoreCompletedTransactionsFailedWithError:(NSError *)error {
    if (flutterResult != nil) {
        flutterResult([FlutterError
                       errorWithCode:[self standardErrorCode:(int)error.code]
                       message:[self englishErrorCodeDescription:(int)error.code]
                       details:nil]);
    }
    flutterResult = nil;
}
```

That will send a PlatformException back to Dart which you will need to handle.  Or you might prefer to check the error code on the iOS side and swallow the error if it was due to the user cancelling.

**Comment by hyochan** [comment]
`2.0.0+` has raised today. Please feel free to reopen after trying out `2.0.0+5` or above.

**Comment by niusounds** [comment]
`restoreCompletedTransactionsFailedWithError` is still not implemented at `2.0.5`.
`getPurchaseHistory()` on iOS never throw error and also never return null.
